modules/admin/proc-running-rhdh-behind-a-proxy.adoc

```diff
@@ -0,0 +1,115 @@
+
+For Helm-based deployment, both a developer or a cluster administrator, with permissions to create resources in the cluster, can configure the proxy variables in a `values.yaml` Helm configuration file.
+
+For Operator-based deployment, the approach you use for proxy configuration is based on your role:
+
+* As a cluster administrator with access to the Operator namespace, you can configure the proxy variables in the Operator's default ConfigMap file. This configuration applies the proxy settings to all the users within the namespace where the Operator is deployed.
+* As a developer, you can configure the proxy variables in a custom resource (CR) file. This configuration applies the proxy settings only to the users with access to that CR.
+
+
+.Prerequisites
+
+* You have installed the {product}. For more information, see xref:proc-install-rhdh_rhdh-getting-started[].
+* You have created a project in the {product-short}.
+
+.Procedure
+
+. Configure proxy information in one of the following deployments based on your needs:
++
+.Helm deployment
+
+* Set the proxy information in your Helm configuration file:
++
+[source,yaml]
+----
+upstream:
+  backstage:
+    extraEnvVars:
+      - name: HTTP_PROXY
+        value: '<http_proxy_url>'
+      - name: HTTPS_PROXY
+        value: '<https_proxy_url>'
+      - name: NO_PROXY
+        value: '<no_proxy_settings>'
+----
++
+Where,
+
+`<http_proxy_url>`:: Denotes a variable that you must replace with the HTTP proxy URL.
+`<https_proxy_url>`:: Denotes a variable that you must replace with the HTTPS proxy URL.
+`<no_proxy_settings>`:: Denotes a variable that you must replace with comma-separated URLs, which you want to exclude from proxying, for example, `foo.com,baz.com`.
++
+.Example: Setting proxy variables using Helm Chart
+
+[source,yaml]
+----
+upstream:
+  backstage:
+    extraEnvVars:
+      - name: HTTP_PROXY
+        value: 'http://10.10.10.105:3128'
+      - name: HTTPS_PROXY
+        value: 'http://10.10.10.106:3128'
+      - name: NO_PROXY
+        value: 'localhost,example.org'
+----
+
++
+.Operator deployment 
+
+** Perform one of the following steps based on your role:
+* As an administrator, set the proxy information in the Operator's default ConfigMap file:
++
+.. Search for a ConfigMap file named `backstage-default-config` in the namespace where the Operator is deployed and open it. 
+.. Find the `deployment.yaml` key.
+.. Set the value of the `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY` environment variables in the `Deployment` spec as shown in the following example:
++ 
+.Example: Setting proxy variables in a ConfigMap file
+[source,yaml]
+----
+# Other fields omitted
+  deployment.yaml: |-
+    apiVersion: apps/v1
+    kind: Deployment
+    # Other fields omitted
```

Comment:
Even if the other fields are omitted, do you want to add the top-level fields as well, to make it clear what the path is? Something like this?
```yaml
# Other fields omitted
  deployment.yaml: |-
    apiVersion: apps/v1
    kind: Deployment
    spec:
      template:
        spec:
          initContainers:
            - name: install-dynamic-plugins
              env:
                - name: NPM_CONFIG_USERCONFIG
                  value: /opt/app-root/src/.npmrc.dynamic-plugins
                - name: HTTP_PROXY
                  value: 'http://10.10.10.105:3128'
                - name: HTTPS_PROXY
                  value: 'http://10.10.10.106:3128'
                - name: NO_PROXY
                  value: 'localhost,example.org'
          containers:
            - name: backstage-backend
              env:
                - name: APP_CONFIG_backend_listen_port
                  value: "7007"
                - name: HTTP_PROXY
                  value: 'http://10.10.10.105:3128'
                - name: HTTPS_PROXY
                  value: 'http://10.10.10.106:3128'
                - name: NO_PROXY
                  value: 'localhost,example.org'
```